assets/js/blocks/attribute-filter/block.js

```diff
@@ -268,42 +268,47 @@ const AttributeFilterBlock = ( {
 		[ pageUrl, attributeObject?.taxonomy ]
 	);
 
-	const onClickSubmit = ( isChecked ) => {
+	const onSubmit = ( checkedFilters ) => {
 		const query = updateAttributeFilter(
 			productAttributesQuery,
 			setProductAttributesQuery,
 			attributeObject,
-			getSelectedTerms( isChecked ),
+			getSelectedTerms( checkedFilters ),
 			blockAttributes.queryType === 'or' ? 'in' : 'and'
 		);
 
 		// This is for PHP rendered template filtering only.
 		if ( filteringForPhpTemplate ) {
-			redirectPageForPhpTemplate( query, isChecked.length === 0 );
+			redirectPageForPhpTemplate( query, checkedFilters.length === 0 );
 		}
```

Comment:
On the shop page with one Filter Products by Attribute block (with filter button enabled). If we comment out these lines, the page is still redirected when clicking the filter button.